lib/puppet/type/component.rb

```diff
@@ -49,6 +49,11 @@ def ref
     reference.to_s
   end
 
+  # This allows us to grab the defined type 'type' for optional use in other custom type's autorelation methods
+  def title_class
```

Comment:
reference is a private method inside of this class I believe.  The name can easily be changed, I wasn't completely sure how to describe it in a way that would be useful.  Something like title_type or something.  Because technically it's grabbing the sudo-type created using component.rb which is what we'd need in order to make this work.  
I'm open to any suggestions and modifications.  @hunner had mentioned this kind of addition might be useful in core and I was doing the 'hack' as mentioned in PUP-5360 and we had decided it might be nice to make that easier to use for other people.